docs/Activation-Recovery.md

```diff
@@ -0,0 +1,234 @@
+
+The Recovery Code and PUK can be obtained in two basic ways:
+
+1. Via the Recovery Postcard
+   - Postcard typically contains one Recovery Code and multiple PUK values.
+   - User has to use PUKs from postcard in the predefined order.
+   - Each PUK can be used only for once.
+   - The Recovery Code must be confirmed before it can be used for a recovery.
+   - See [Recovery Postcard](Recovery-Postcard.md) document for more details.
+
+1. As a part of an application activation
+   - In this case, only one pair of Recovery Code and PUK is generated
+   - This type of Recovery Code is associated to the activation and if it's used for the recovery, then the previous activation is removed.
+   - This type of Recovery Code and PUK can be obtained via any type of the activation process.
+   - See [Application provided Recovery Code and PUK](#application-provided-recovery-code-and-puk) chapter for more details.
+
+
+
+## Implications to security
+
+The Activation via Recovery Code is by default **disabled** on PowerAuth Server. Before you enable and configure this feature, you should read this chapter and decide, whether it is appropriate and safe for your purposes.
+
+### Application is active from the beginning
+
+If application is activated via the Recovery Code and PUK, then it goes immediately to **ACTIVE** state (see [Activation States](Activation.md#activation-states)). There's no intermediate state when the user can confirm that activation in the additional digital channel (like internet banking). This fact limits this feature mostly to only limited application scenarios:
+
+1. Only for the application that doesn't provide active operations
+2. The application itself acts as a confirmation device (like our [Mobile Token](https://www.wultra.com/product/powerauth-mobile-token) does)
+
+For the first case, we don't recommend to allow this feature for PowerAuth powered applications, that provide "active" operations. That's typically a mobile banking app that can withdraw money from user's account. The reason for that is that if the Recovery Code and PUK are compromised, then the attacker can easily harm the user.
+
+In the second scenario, the mobile application acts as a replacement for a typical authorization channel, like SMS, or another hardware token. In this situation, compromising Recovery Code and PUK can be still very dangerous, but requires a more coordinated attack to harm the user. The attacker has to typically gain an access to additional credentials, like login and password to internet banking.
+
+For all cases, we recommend you to implement the following countermeasures:
+
+- Your application should receive a push notification once the activation is recovered on another device.
+- You should adequately inform the user about how sensitive Recovery Code and PUK are.
```

Comment:
I would also suggest here using some other channel for notifications, such as sending a customer an e-mail or SMS message...